packages/tas_keybind/init.lua

```diff
@@ -0,0 +1,72 @@
+tas = tas or {}
+local oldPlayerControls = {}
+---@type table<tas.CONTROLS, Set<tas.ControlListenerCallback>>
+local controlListeners = {}
+
+---@alias tas.ControlListenerCallback fun(player: unknown, pressed: boolean):void
+---@alias Set<T> table<T, boolean>
+
+---@enum tas.CONTROLS
+tas.CONTROLS = {
+	up = 0,
+	down = 1,
+	left = 2,
+	right = 3,
+	jump = 4,
+	aux1 = 5,
+	sneak = 6,
+	dig = 7,
+	place = 8,
+	zoom = 9,
+}
+
+for _, control in pairs(tas.CONTROLS) do
+	controlListeners[control] = {}
+end
+
+local function getControlChanges(playerName, newControls)
+	local oldControls = oldPlayerControls[playerName] or 0
+	oldPlayerControls[playerName] = newControls
+
+	return bit.bxor(newControls, oldControls)
+end
+
+local function callControlListeners(player)
+	local newControls = player:get_player_control_bits()
+	local changedControls = getControlChanges(player:get_player_name(), newControls)
+
+	for controlName, controlBitIndex in pairs(tas.CONTROLS) do
+		local controlBitmask = bit.rshift(1, controlBitIndex)
```

Comment:
Shouldn't this be a left shift?

Reply:
Why would it be a left shift? I mean unless I'm forgetting basic bitwise operations, right shifting 1 by N makes a binary number where a 1 is at N and 0 for everything else.

Reply:
```
bitfield = 011101010110111010
           <-left- 1 -right->
```